src/Lookup/UnresolvedEntityRedirectException.php

```diff
@@ -23,9 +24,15 @@ class UnresolvedEntityRedirectException extends EntityLookupException {
 	/**
 	 * @param EntityId $entityId
 	 * @param EntityId $redirectTargetId The ID of the target Entity of the redirect
+	 * @param string|null $message
+	 * @param Exception|null $previous
 	 */
-	public function __construct( EntityId $entityId, EntityId $redirectTargetId ) {
-		parent::__construct( $entityId, "Unresolved redirect to " . $redirectTargetId->getSerialization() );
+	public function __construct( EntityId $entityId, EntityId $redirectTargetId, $message = null, Exception $previous = null ) {
+		parent::__construct(
+			$entityId,
+			$message !== null ? $message : "Unresolved redirect to " . $redirectTargetId->getSerialization(),
```

Comment:
> So the two parameters $redirectTargetId and $message are now exclusive?

No. The message that will be constructed when none is provided uses $redirectTargetId. That is as far as the relation between those two parameters goes.

> Or to ask this via a different question: What's the use of a custom message?

To allow the things throwing the extension to provide a more specific error message. Same reason as why Exception itself has a message parameter.